### PR TITLE
Add WSO2 EI Analytics Solution

### PR DESCRIPTION
**Pull request by tharindu-bandara**

### CODE REVIEW

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/components/siddhi-extensions/siddhi-execution-esbanalytics/src/test/java/org/wso2/extension/siddhi/execution/esbanalytics/TestCaseOfDecompressStreamProcessorExtension.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import org.wso2.siddhi.core.SiddhiAppRuntime;
+import org.wso2.siddhi.core.SiddhiManager;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.core.stream.output.StreamCallback;
+import org.wso2.siddhi.core.util.SiddhiTestHelper;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class TestCaseOfDecompressStreamProcessorExtension {
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/components/siddhi-extensions/siddhi-execution-esbanalytics/src/test/java/org/wso2/extension/siddhi/execution/esbanalytics/TestCaseOfDecompressStreamProcessorExtension.java]: Shall we rename this class to something like `DecompressStreamProcessorTest` ? Please refer how other Test classes are named in siddhi extensions.

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/components/siddhi-extensions/siddhi-execution-esbanalytics/src/test/java/org/wso2/extension/siddhi/execution/esbanalytics/TestCaseOfDecompressStreamProcessorExtension.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    private static Logger log = Logger.getLogger(TestCaseOfDecompressStreamProcessorExtension.class);
+    private AtomicInteger count = new AtomicInteger(0);
+    private volatile boolean decompressedAttributesRetrieved;
+
+    @BeforeMethod
+    public void init() {
+
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/components/siddhi-extensions/siddhi-execution-esbanalytics/src/test/java/org/wso2/extension/siddhi/execution/esbanalytics/TestCaseOfDecompressStreamProcessorExtension.java]: remove empty space

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/components/siddhi-extensions/siddhi-execution-esbanalytics/src/test/java/org/wso2/extension/siddhi/execution/esbanalytics/TestCaseOfDecompressStreamProcessorExtension.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        count.set(0);
+        decompressedAttributesRetrieved = false;
+    }
+
+    @Test
+    public void testDecompressSimpleMessage() throws InterruptedException {
+
+        log.info("DecompressSimpleMessage TestCase");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        //siddhiManager.setExtension("str:concat", ConcatFunctionExtension.class);
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/components/siddhi-extensions/siddhi-execution-esbanalytics/src/test/java/org/wso2/extension/siddhi/execution/esbanalytics/TestCaseOfDecompressStreamProcessorExtension.java]: remove this line.

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/components/siddhi-extensions/siddhi-execution-esbanalytics/src/test/java/org/wso2/extension/siddhi/execution/esbanalytics/util/ESBAnalyticsDecompressConstants.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        "V5traysNy0avXq3MK1owv+AOj9MLYVZ2tlm2rF5BQdBeYMWW+RzQOJlyXOein5K4K6VARRF6DjBJQDNFoCHKzsc3IspmilrPuks" +
+        "eN2lobDXIoRIpXYD3seK4lAZ0XNkGSjpLAfLDX/C3uUVlmhS6twMqSClZFT5lqaRKWrq1BuO7oDYuRedaFJMc32UacXxChS90GR" +
+        "lART0r59/ik1fJByOgsiNXOQknLo8XZx6nhzSB1xaMShEYeOA4e0fb7pZEDVD0Tl/gFHOmeyINKyRzqPMzui8eNNIm1EohGJRiQ" +
+        "6BiTCPe/3vVsiGdD5aaL+4B/czklVFSUAAA==";
+    }
+
+    public static void main(String[] args) {
+
+        System.out.println(TestData.FLOW_DATA);
+    }
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/components/siddhi-extensions/siddhi-execution-esbanalytics/src/test/java/org/wso2/extension/siddhi/execution/esbanalytics/util/ESBAnalyticsDecompressConstants.java]: Do we need this main method?

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/HorizontalBarChart/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var path = require('path');
+var webpack = require('webpack');
+var CopyWebpackPlugin = require('copy-webpack-plugin');
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/HorizontalBarChart/webpack.config.js]: should be `copyWebpackPlugin`

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MediatorProperties/src/dataTables.bootstrap.css [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+table.table-bordered th:last-child,
+table.table-bordered td:last-child {
+	border-right-width: 0;
+}
+div.dataTables_scrollHead table.table-bordered {
+	border-bottom-width: 0;
+}
+
+
+
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MediatorProperties/src/dataTables.bootstrap.css]: remove unnecessary linespaces

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MediatorProperties/src/dataTables.bootstrap.css [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+    background: -webkit-gradient(linear, left top, right top, color-stop(0%, rgba(255,255,255,0)), color-stop(25%, rgba(255,255,255,0.9)), color-stop(75%, rgba(255,255,255,0.9)), color-stop(100%, rgba(255,255,255,0)));
+    background: -webkit-linear-gradient(left, rgba(255,255,255,0) 0%, rgba(255,255,255,0.9) 25%, rgba(255,255,255,0.9) 75%, rgba(255,255,255,0) 100%);
+    background: -moz-linear-gradient(left, rgba(255,255,255,0) 0%, rgba(255,255,255,0.9) 25%, rgba(255,255,255,0.9) 75%, rgba(255,255,255,0) 100%);
+    background: -ms-linear-gradient(left, rgba(255,255,255,0) 0%, rgba(255,255,255,0.9) 25%, rgba(255,255,255,0.9) 75%, rgba(255,255,255,0) 100%);
+    background: -o-linear-gradient(left, rgba(255,255,255,0) 0%, rgba(255,255,255,0.9) 25%, rgba(255,255,255,0.9) 75%, rgba(255,255,255,0) 100%);
+    background: linear-gradient(to right, rgba(255,255,255,0) 0%, rgba(255,255,255,0.9) 25%, rgba(255,255,255,0.9) 75%, rgba(255,255,255,0) 100%);
+}
+
+
+
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MediatorProperties/src/dataTables.bootstrap.css]: remove unnecessary linespaces

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MediatorProperties/utils/CodeMirror/addon/merge/merge.js [hunk 8]
```diff
@@ -0,0 +1,3176 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// declare global: diff_match_patch, DIFF_INSERT, DIFF_DELETE, DIFF_EQUAL
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MediatorProperties/utils/CodeMirror/addon/merge/merge.js]: remove comment

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/aggregatemediator.svg [hunk 9]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ ~   Copyright (c) 2017 WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ ~
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/aggregatemediator.svg]: fix year.

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/bammediator.svg [hunk 10]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ ~   Copyright (c) 2017 WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/bammediator.svg]: fix year

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/beanmediator.svg [hunk 11]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ ~   Copyright (c) 2017 WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/beanmediator.svg]: fix year

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/cachemediator.svg [hunk 12]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ ~   Copyright (c) 2017 WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/cachemediator.svg]: fix year

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/callmediator.svg [hunk 13]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ ~   Copyright (c) 2017 WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/callmediator.svg]: fix year

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/calloutmediator.svg [hunk 14]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ ~   Copyright (c) 2017 WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/calloutmediator.svg]: fix year

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/calltemplatemediator.svg [hunk 15]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ ~   Copyright (c) 2017 WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/calltemplatemediator.svg]: fix year

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/classmediator.svg [hunk 16]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ ~   Copyright (c) 2017 WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/classmediator.svg]: fix year

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/clonemediator.svg [hunk 17]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ ~   Copyright (c) 2017 WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/clonemediator.svg]: fix year here and in all svg files, if applicable.

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/SearchBox/webpack.config.js [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const path = require('path');
+const webpack = require('webpack');
+const CopyWebpackPlugin = require('copy-webpack-plugin');
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/SearchBox/webpack.config.js]: fix name

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/StatsChart/webpack.config.js [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const path = require('path');
+const webpack = require('webpack');
+const CopyWebpackPlugin = require('copy-webpack-plugin');
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/StatsChart/webpack.config.js]: fix name

File: features/sp-solution-features/org.wso2.carbon.sp.solutions.ei.analytics.feature/pom.xml [hunk 20]
```diff
@@ -0,0 +1,182 @@
...
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <name>WSO2 Stream Processor - EI Analytics Solution Feature</name>
+    <artifactId>org.wso2.carbon.sp.solutions.ei.analytics.feature</artifactId>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.wso2.analytics.solutions</groupId>
+            <artifactId>wso2sp-solutions.ei-analytics.statschart</artifactId>
+            <version>1.0.0-SNAPSHOT</version>
```
Comment (dilini-muthumala) [inline features/sp-solution-features/org.wso2.carbon.sp.solutions.ei.analytics.feature/pom.xml]: remove versions here and in other dependencies as well.

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/components/siddhi-extensions/siddhi-execution-esbanalytics/src/test/java/org/wso2/extension/siddhi/execution/esbanalytics/TestCaseOfDecompressStreamProcessorExtension.java (hunk 2 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/components/siddhi-extensions/siddhi-execution-esbanalytics/src/test/java/org/wso2/extension/siddhi/execution/esbanalytics/TestCaseOfDecompressStreamProcessorExtension.java]: This comes with code style formatting.

File: features/sp-solution-features/org.wso2.carbon.sp.solutions.ei.analytics.feature/pom.xml (hunk 20 above)
Comment (tharindu-bandara) [reply features/sp-solution-features/org.wso2.carbon.sp.solutions.ei.analytics.feature/pom.xml]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/StatsChart/webpack.config.js (hunk 19 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/StatsChart/webpack.config.js]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/SearchBox/webpack.config.js (hunk 18 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/SearchBox/webpack.config.js]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/clonemediator.svg (hunk 17 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/clonemediator.svg]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/classmediator.svg (hunk 16 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/classmediator.svg]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/calltemplatemediator.svg (hunk 15 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/calltemplatemediator.svg]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/calloutmediator.svg (hunk 14 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/calloutmediator.svg]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/callmediator.svg (hunk 13 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/callmediator.svg]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/cachemediator.svg (hunk 12 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/cachemediator.svg]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/beanmediator.svg (hunk 11 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/beanmediator.svg]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/bammediator.svg (hunk 10 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/bammediator.svg]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/aggregatemediator.svg (hunk 9 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MessageFlow/mediators/aggregatemediator.svg]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MediatorProperties/utils/CodeMirror/addon/merge/merge.js (hunk 8 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MediatorProperties/utils/CodeMirror/addon/merge/merge.js]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MediatorProperties/src/dataTables.bootstrap.css (hunk 7 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MediatorProperties/src/dataTables.bootstrap.css]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MediatorProperties/src/dataTables.bootstrap.css (hunk 6 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/MediatorProperties/src/dataTables.bootstrap.css]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/HorizontalBarChart/webpack.config.js (hunk 5 above)
Comment (tharindu-bandara) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/widgets/HorizontalBarChart/webpack.config.js]: Fixed in 53951238f84729f0c325006f8def576a3c702eda

File: components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/components/siddhi-extensions/siddhi-execution-esbanalytics/src/test/java/org/wso2/extension/siddhi/execution/esbanalytics/TestCaseOfDecompressStreamProcessorExtension.java (hunk 2 above)
Comment (Niveathika92) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.ei.analytics/components/siddhi-extensions/siddhi-execution-esbanalytics/src/test/java/org/wso2/extension/siddhi/execution/esbanalytics/TestCaseOfDecompressStreamProcessorExtension.java]: AFAIK empty spaces are not removed by code style formatting. Shall we remove manually? Or the code style formatting enforces this?

**Comment by Niveathika92** [comment]
Standardise the widget names such that they start as EIAnalytics..... (Directory, main class file etc..)

File: components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/HorizontalBarChart/pom.xml [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>wso2sp-solutions.ei-analytics.horizontalbarchart</artifactId>
```
Comment (Niveathika92) [inline components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/HorizontalBarChart/pom.xml]: Rename the widget with meaningful name

File: components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/HorizontalBarChart/webpack.config.js [hunk 22]
```diff
@@ -0,0 +1,81 @@
...
+        ],
+    },
+    plugins: [
+        new copyWebpackPlugin([
+            {from: path.resolve(__dirname, './src/resources/')},
+        ]),
+    ],
+    resolve: {
+        extensions: ['.js', '.json', '.jsx', '.scss'],
+    },
```
Comment (Niveathika92) [inline components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/HorizontalBarChart/webpack.config.js]: externalise React 
https://github.com/wso2/analytics-solutions/blob/master/components/sp-solutions/org.wso2.carbon.sp.solutions.http.analytics/widgets/HTTPAnalyticsLatencyComparison/webpack.config.js#L91

File: components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/MediatorProperties/pom.xml [hunk 23]
```diff
@@ -0,0 +1,169 @@
...
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>wso2sp-solutions.ei-analytics.mediatorproperties</artifactId>
```
Comment (Niveathika92) [inline components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/MediatorProperties/pom.xml]: Follow standard org.wso2.carbon.analytics....... . See sp-solutions naming

File: components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/MediatorProperties/src/resources/widgetConf.json [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+    "pubsub": {
+      "types": [
+        "subscriber"
+      ]
+    },
+    "providerConfig": {
+      "configs": {
+        "type": "SiddhiStoreDataProvider",
+        "config": {
+          "siddhiApp": "@App:name('ESBEventTable') @store(type = 'rdbms', datasource = 'EI_ANALYTICS', field.length='contextPropertyMap:5000,beforePayload:5000,afterPayload:5000,transportPropertyMap:5000' ) @Index('metaTenantId','messageFlowId','host','hashCode','componentName','componentType','componentIndex','componentId','startTime','endTime','entryPoint','entryPointHashcode','faultCount') define table ESBEventTable (metaTenantId int, messageFlowId string, host string, hashCode string, componentName string, componentType string, componentIndex int, componentId string, startTime long, endTime long, duration long, beforePayload string, afterPayload string, contextPropertyMap string, transportPropertyMap string, children string, entryPoint string, entryPointHashcode string, faultCount int, eventTimestamp long);",
```
Comment (Niveathika92) [inline components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/MediatorProperties/src/resources/widgetConf.json]: Use the same siddhi app name.. for easy debugging later on. However, add only the definitions needed. (Event table/Aggregations) remove all processing logic

Comment (Niveathika92) [reply components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/MediatorProperties/src/resources/widgetConf.json]: Ignore the comment on definitions. seems only table definitions are their

File: components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/HorizontalBarChart/pom.xml (hunk 21 above)
Comment (tharindu-bandara) [reply components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/HorizontalBarChart/pom.xml]: Fixed with ( 8d4a31838382c2521640a73fbada3caf4c73101b 7b2df2b095d7ce643a2b920943ee88cfbab0d3bc 96d518077e89317005fe59ff8670a05d4f705b4b 9ae3d822e7addb1c50a2d7603489a4dd64be3180 07bb8f914fefa6f383f6b7ffce467290a322e356 d40163630eec02a0bee35aec75ef149842b4f45e 0df301308a5496995274a95eed4b528881a5e484 )

File: components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/HorizontalBarChart/webpack.config.js (hunk 22 above)
Comment (tharindu-bandara) [reply components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/HorizontalBarChart/webpack.config.js]: Fixed with ( 8d4a31838382c2521640a73fbada3caf4c73101b 7b2df2b095d7ce643a2b920943ee88cfbab0d3bc 96d518077e89317005fe59ff8670a05d4f705b4b 9ae3d822e7addb1c50a2d7603489a4dd64be3180 07bb8f914fefa6f383f6b7ffce467290a322e356 d40163630eec02a0bee35aec75ef149842b4f45e 0df301308a5496995274a95eed4b528881a5e484 )

File: components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/MediatorProperties/pom.xml (hunk 23 above)
Comment (tharindu-bandara) [reply components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/MediatorProperties/pom.xml]: Fixed with ( 8d4a31838382c2521640a73fbada3caf4c73101b 7b2df2b095d7ce643a2b920943ee88cfbab0d3bc 96d518077e89317005fe59ff8670a05d4f705b4b 9ae3d822e7addb1c50a2d7603489a4dd64be3180 07bb8f914fefa6f383f6b7ffce467290a322e356 d40163630eec02a0bee35aec75ef149842b4f45e 0df301308a5496995274a95eed4b528881a5e484 )

File: components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/MediatorProperties/src/resources/widgetConf.json (hunk 24 above)
Comment (tharindu-bandara) [reply components/ei-solutions/org.wso2.carbon.ei.solutions.ei.analytics/widgets/MediatorProperties/src/resources/widgetConf.json]: Fixed with ( 8d4a31838382c2521640a73fbada3caf4c73101b 7b2df2b095d7ce643a2b920943ee88cfbab0d3bc 96d518077e89317005fe59ff8670a05d4f705b4b 9ae3d822e7addb1c50a2d7603489a4dd64be3180 07bb8f914fefa6f383f6b7ffce467290a322e356 d40163630eec02a0bee35aec75ef149842b4f45e 0df301308a5496995274a95eed4b528881a5e484 )

File: pom.xml [hunk 25]
```diff
@@ -251,6 +253,65 @@
                 <version>${project.version}</version>
             </dependency>
 
+            <!--Ei Analytics solution dependencies-->
+            <dependency>
+                <groupId>org.wso2.analytics.solutions</groupId>
+                <artifactId>wso2sp-solutions.ei-analytics.statschart</artifactId>
+                <version>1.0.0-SNAPSHOT</version>
```
Comment (dilini-muthumala) [inline pom.xml]: Please use ${project.version} instead of `1.0.0-SNAPSHOT` . Please fix in other places as well.

File: pom.xml [hunk 26]
```diff
@@ -489,6 +550,8 @@
 
     <properties>
         <carbon.analytics-common.version>6.0.66</carbon.analytics-common.version>
+        <carbon.analytics.spark.core.version>1.2.3</carbon.analytics.spark.core.version>
```
Comment (dilini-muthumala) [inline pom.xml]: Is this needed?

File: pom.xml (hunk 25 above)
Comment (tharindu-bandara) [reply pom.xml]: Fixed in aa6fe88acd5407ba41e69cbcce5664b7c38f8773

File: pom.xml (hunk 26 above)
Comment (tharindu-bandara) [reply pom.xml]: It's required for decompress extension.

Comment (dilini-muthumala) [reply pom.xml]: ok.

**Pull request merged**